financial-services-accelerator/components/org.wso2.financial.services.accelerator.gateway/src/main/java/org/wso2/financial/services/accelerator/gateway/executor/impl/error/handling/DefaultErrorHandlingExecutor.java

```diff
@@ -0,0 +1,181 @@
+        HashSet<String> statusCodes = new HashSet<>();
+
+        for (FSExecutorError error : errors) {
+            statusCodes.add(error.getHttpStatusCode());
+        }
+
+        payload.put(ERRORS_TAG, errorList);
+        if (errorList.length() != 0) {
+            fsApiRequestContext.setModifiedPayload(payload.toString());
+            Map<String, String> addedHeaders = fsApiRequestContext.getAddedHeaders();
+            addedHeaders.put(GatewayConstants.CONTENT_TYPE_TAG, GatewayConstants.JSON_CONTENT_TYPE);
+            fsApiRequestContext.setAddedHeaders(addedHeaders);
+        }
+        int statusCode;
+        if (fsApiRequestContext.getContextProps().containsKey(GatewayConstants.ERROR_STATUS_PROP)) {
+            statusCode = Integer.parseInt(fsApiRequestContext
+                    .getContextProperty(GatewayConstants.ERROR_STATUS_PROP).toString());
+        } else if (isAnyClientErrors(statusCodes)) {
+            statusCode = HttpStatus.SC_BAD_REQUEST;
+        } else {
+            statusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+        }
+        fsApiRequestContext.addContextProperty(GatewayConstants.ERROR_STATUS_PROP,
+                String.valueOf(statusCode));
+    }
+
+    private void handleResponseError(FSAPIResponseContext fsApiResponseContext) {
+
+        if (!fsApiResponseContext.isError()) {
+            return;
+        }
+        JSONObject payload = new JSONObject();
+        ArrayList<FSExecutorError> errors = fsApiResponseContext.getErrors();
+        JSONArray errorList = getErrorJSON(errors);
+        HashSet<String> statusCodes = new HashSet<>();
+
+        for (FSExecutorError error : errors) {
+            statusCodes.add(error.getHttpStatusCode());
+        }
+
+        payload.put(ERRORS_TAG, errorList);
+        fsApiResponseContext.setModifiedPayload(payload.toString());
+        Map<String, String> addedHeaders = fsApiResponseContext.getAddedHeaders();
+        addedHeaders.put(GatewayConstants.CONTENT_TYPE_TAG, GatewayConstants.JSON_CONTENT_TYPE);
+        fsApiResponseContext.setAddedHeaders(addedHeaders);
+        int statusCode;
+        if (fsApiResponseContext.getContextProps().containsKey(GatewayConstants.ERROR_STATUS_PROP)) {
+            statusCode = Integer.parseInt(fsApiResponseContext
+                    .getContextProperty(GatewayConstants.ERROR_STATUS_PROP).toString());
+        } else if (isAnyClientErrors(statusCodes)) {
+            statusCode = HttpStatus.SC_BAD_REQUEST;
+        } else {
+            statusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+        }
+        fsApiResponseContext.addContextProperty(GatewayConstants.ERROR_STATUS_PROP,
+                String.valueOf(statusCode));
+    }
+
+    private JSONArray getErrorJSON(List<FSExecutorError> errors) {
+
+        JSONArray errorList = new JSONArray();
+        for (FSExecutorError error : errors) {
+            JSONObject errorObj = new JSONObject();
+            errorObj.put("code", error.getCode());
```

Comment:
Constants?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/130/commits/1e13ec5af065388311c13d8ca7f04ef804d84184